File: src/main/java/com/enterprise/crm/v1/customer/service/CustomerService.java
package com.enterprise.crm.v1.customer.service;

import com.enterprise.crm.v1.customer.dto.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDateTime;
import java.util.UUID;

public interface CustomerService {
    CustomerResponse createCustomer(CreateCustomerRequest request);
    CustomerResponse updateCustomer(UUID id, CreateCustomerRequest request);
    CustomerResponse getCustomerById(UUID id);
    Page<CustomerResponse> searchCustomers(
            String companyName,
            String customerStatus,
            UUID assignedSalesRepId,
            String tag,
            LocalDateTime startCreated,
            LocalDateTime endCreated,
            LocalDateTime startUpdated,
            LocalDateTime endUpdated,
            String companySize,
            String taxIdentifier,
            Pageable pageable
    );
    void deleteCustomer(UUID id);
    void restoreCustomer(UUID id);
    void assignCustomer(UUID id, UUID assignedSalesRepId);
}

File: frontend/src/services/customerService.js
import api from './api';

const customerService = {
  createCustomer: async (customerData) => {
    return await api.post('/api/v1/customers', customerData);
  },

  updateCustomer: async (id, customerData) => {
    return await api.put(`/api/v1/customers/${id}`, customerData);
  },

  getCustomerById: async (id) => {
    return await api.get(`/api/v1/customers/${id}`);
  },

  searchCustomers: async (params = {}) => {
    return await api.get('/api/v1/customers', { params });
  },

  deleteCustomer: async (id) => {
    return await api.delete(`/api/v1/customers/${id}`);
  },

  restoreCustomer: async (id) => {
    return await api.put(`/api/v1/customers/${id}/restore`);
  },

  assignCustomer: async (id, assignedSalesRepId) => {
    return await api.put(`/api/v1/customers/${id}/assign`, { assignedSalesRepId });
  },
};

export default customerService;
